frontend/src/components/account/AccountPage.js

```diff
@@ -193,6 +208,91 @@ export default function AccountPage({
     </Link>
   );
 
+  // Lines 210 - 300 handle follow button for organizations
+  const [confirmDialogOpen, setConfirmDialogOpen] = useState({
+    follow: false,
+  });
+
+  const onFollowDialogClose = (confirmed) => {
+    if (confirmed) toggleFollowOrganization();
+    setConfirmDialogOpen({ ...confirmDialogOpen, follow: false });
+  };
+
+  const { showFeedbackMessage } = useContext(FeedbackContext);
+
+  const handleToggleFollowOrganization = () => {
+    if (!token)
+      showFeedbackMessage({
+        message: <span>{organizationTexts.please_log_in_to_follow_an_organization}</span>,
+        error: true,
+        promptLogIn: true,
+      });
+    else if (isUserFollowing) setConfirmDialogOpen({ ...confirmDialogOpen, follow: true });
+    else toggleFollowOrganization();
+  };
+
+  const toggleFollowOrganization = () => {
+    handleFollow(isUserFollowing, false, true);
+    apiRequest({
+      method: "post",
+      url: "/api/organizations/" + account.url_slug + "/set_follow/",
+      payload: { following: !isUserFollowing },
+      token: token,
+      locale: locale,
+    })
+      .then(function (response) {
+        handleFollow(response.data.following, true, false);
+        updateFollowers();
+        showFeedbackMessage({
+          message: response.data.message,
+        });
+      })
+      .catch(function (error) {
+        console.log(error);
+        if (error && error.reponse) console.log(error.response);
+      });
+  };
+
+  const handleReadNotifications = async (notificationType) => {
+    const notification_to_set_read = notifications.filter(
+      (n) =>
+        n.notification_type === notificationType && n.organization.url_slug === account.url_slug
+    );
+    await setNotificationsRead(token, notification_to_set_read, locale);
+    await refreshNotifications();
+  };
+
+  const { notifications, setNotificationsRead, refreshNotifications } = useContext(UserContext);
+
+  const [initiallyCaughtFollowers, setInitiallyCaughtFollowers] = useState(false);
+  const [followers, setFollowers] = useState([]);
+  const [showFollowers, setShowFollowers] = useState(false);
+  const toggleShowFollowers = async () => {
+    setShowFollowers(!showFollowers);
+    if (!initiallyCaughtFollowers) {
+      await updateFollowers();
+      handleReadNotifications(NOTIFICATION_TYPES.indexOf("organization_follower"));
+      setInitiallyCaughtFollowers(true);
+    }
+  };
+  const updateFollowers = async () => {
+    const retrievedFollowers = await getFollowers(account, token, locale);
+    setFollowers(retrievedFollowers);
+  };
+
+  const [gotParams, setGotParams] = useState(false);
+  useEffect(() => {
+    if (!gotParams) {
+      const params = getParams(window.location.href);
+      if (params.show_followers && !showFollowers) toggleShowFollowers();
+      setGotParams(true);
+    }
+  });
+
+  const bindFollow = useLongPress(() => {
+    toggleShowFollowers();
+  });
+
```

Comment:
Would it be possible/make sense to hide all this complexity (or at least the part that doesn't affect the rest of the page) in the `FollowButton`component? I'm not sure myself, generally we'll have to start moving stuff outside of the AccountPage eventually.

Reply:
Or it could be an inbetween-component `OrganizationFollowButton`. But again: not sure if this makes sense

Reply:
What's the status on this one again?

Reply:
We had discussed this & decided to leave this as is for the moment.

Reply:
:+1: